op-service/eth/transactions.go

```diff
@@ -127,25 +117,43 @@ type GenericTx interface {
 	// This will return types.ErrTxTypeNotSupported if the transaction-type is not supported.
 	Transaction() (*types.Transaction, error)
 
-	// TxType returns the EIP-2718 type.
+	// TxType returns the EIP-2718 type, or an error if the transaction is not an EIP-2718 transaction.
 	TxType() uint8
```

Comment:
I think it would be better to add a sentinel error, like `ErrInvalidTxType` to the L1Fetcher implementations that are returned by `InfoAndTxsByHash` so that a `GenericTx` is always guaranteed to have a `TxType` and a `TxHash`.